<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="marker_size" default="10.8" />
  <arg name="max_new_marker_error" default="0.1" />
  <arg name="max_track_error" default="0.07" />
  <arg name="cam_image_topic" default="/agv2/camera/image_raw" />
  <arg name="cam_info_topic" default="/agv2/camera/camera_info" />
  <arg name="output_frame" default="/agv2/camera_link" />

  <!-- <node pkg="tf" type="static_transform_publisher" name="world_frames_connection" args="0.0 0 0 0 0 0 /camera_link /agv2/camera_link 100" /> -->

  <node name="agv2_ar_track_alvar" pkg="ar_track_alvar" type="individualMarkersNoKinect" respawn="false" output="screen">
    <param name="marker_size" type="double" value="$(arg marker_size)" />
    <param name="max_new_marker_error" type="double" value="$(arg max_new_marker_error)" />
    <param name="max_track_error" type="double" value="$(arg max_track_error)" />
    <param name="output_frame" type="string" value="$(arg output_frame)" />

    <remap from="camera_image" to="$(arg cam_image_topic)" />
    <remap from="camera_info" to="$(arg cam_info_topic)" />
  </node>
</launch>